uwblueprint/focus-on-nature
# feat: add refund link generation

## Notion ticket link

[Cancellation/Refund: Generate a unique cancellation link on a per-registration basis](https://www.notion.so/uwblueprintexecs/Cancellation-Refund-Generate-a-unique-cancellation-link-on-a-per-registration-basis-89c51d3fb72a43bbb958f2d4e9da5ef9)



## Implementation description
* Generated unique refund link by generating refund code for each confirmation email sent to parent after they successfully register for a FON camp
* Created endpoint at `/refund/{refundCode}` that returns the refund information associated with the refund code



## Steps to test
1. Go to http://localhost:3000/register/camp/648df427152c24ef5871842a
2. Fill in the required fields and register some campers using an accessible email
3. Once registration is completed, check the email for an email from FON. It should contain a refund link like the one in screenshot below
4. Copy-paste the refundCode (after /refund) to a text doc
5. In Postman, send a GET request to http://localhost:3000/campers/refund/theRefundCodeFromTextDoc
6. Verify response is correct (array of campers registered for that camp and under each camper the sessions they registered for)
<img width="902" alt="Screen Shot 2023-06-17 at 2 00 09 PM" src="https://github.com/uwblueprint/focus-on-nature/assets/88808428/1f387ad0-4a5b-4605-85b9-d8b8e8b689ed">


## What should reviewers focus on?
* Check there's a refund link with refund code in the url in the confirmation email
* Check the response from the endpoint is correct


## Checklist
